assets/js/amp-customize-controls.js

```diff
@@ -0,0 +1,201 @@
+		 *
+		 * @param {string} url URL.
+		 * @return {string} AMPified URL.
+		 */
+		function setCurrentAmpUrl( url ) {
+			var enabled = api.state( 'ampEnabled' ).get();
+			if ( ! enabled && component.isAmpUrl( url ) ) {
+				return component.unampifyUrl( url );
+			} else if ( enabled && ! component.isAmpUrl( url ) ) {
+				return component.ampifyUrl( url );
+			}
+			return url;
+		}
+
+		/**
+		 * Swap to AMP version of URL in preview.
+		 *
+		 * @return {void}
+		 */
+		function updatePreviewUrl() {
+			api.previewer.previewUrl.set( setCurrentAmpUrl( api.previewer.previewUrl.get() ) );
+		}
+
+		// AMP panel triggers the input toggle for AMP preview.
+		panel.expanded.bind( function() {
+			if ( api.state( 'ampAvailable' ).get() ) {
+				api.state( 'ampEnabled' ).set( panel.expanded.get() );
+			}
+		} );
+
+		// Enable AMP toggle if available and mobile device selected.
+		api.previewedDevice.bind( function( device ) {
+			if ( 'mobile' === device && api.state( 'ampAvailable' ).get() ) {
+				api.state( 'ampEnabled' ).set( true );
+			}
+		} );
+
+		// Message coming from previewer.
+		api.previewer.bind( 'amp-status', function( data ) {
+			api.state( 'ampAvailable' ).set( data.available );
+		} );
+		function setInitialAmpEnabledState( data ) {
+			api.state( 'ampEnabled' ).set( data.enabled );
+			api.previewer.unbind( 'amp-status', setInitialAmpEnabledState );
+		}
+		api.previewer.bind( 'amp-status', setInitialAmpEnabledState );
+
+		// Persist the presence or lack of the amp=1 param when navigating in the preview,
+		// even if current page is not yet supported.
+		api.previewer.previewUrl.validate = ( function( prevValidate ) {
+			return function( value ) {
+				var val = prevValidate.call( this, value );
+				if ( val ) {
+					val = setCurrentAmpUrl( val );
+				}
+				return val;
+			};
+		} )( api.previewer.previewUrl.validate );
+
+		// Listen for ampEnabled state changes.
+		api.state( 'ampEnabled' ).bind( function( enabled ) {
+			$( '.amp-toggle input' ).prop( 'checked', enabled );
+			updatePreviewUrl();
+		} );
+
+		// Listen for ampAvailable state changes.
+		api.state( 'ampAvailable' ).bind( function( available ) {
+			$( '.amp-toggle input' ).prop( 'disabled', ! available );
+		} );
+
+		// Adding checkbox toggle before device selection.
+		$( '.devices-wrapper' ).before( wp.template( 'customize-amp-enabled-toggle' ) );
+
+		// User clicked link within tooltip, go to linked post in preview.
+		$( '.amp-toggle .tooltip a' ).on( 'click', function() {
+			var url = $( this ).data( 'post' );
+			if ( url.length ) {
+				api.state( 'ampEnabled' ).set( true );
+				api.previewer.previewUrl.set( url );
+			}
+		} );
+
+		// Main controls for toggling AMP preview.
+		$( '#customize-footer-actions' ).on( 'click', '.amp-toggle', function() {
+			var $input = $( 'input', $( this ) ),
+				$tooltip = $( '.tooltip', $( this ) ),
+				tooltipTimer = 5000;
+
+			if ( $input.prop( 'disabled' ) ) {
```

Comment:
It would be good to prevent the tooltip from disappearing in the users mouse is over the tooltip.

Reply:
Or when keyboard focus is on the link in the tooltip.